### ISSUE TITLE
6157 - Missing step name pop-up in combat

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#2229]
Got a pop-up while clicking through on a battle.
```
Aug 15, 2017 12:32:01 PM games.strategy.triplea.ui.BattleStepsPanel setStep
INFO: Could not find step name:British withdraw?
```

![clicking_through](https://user-images.githubusercontent.com/12397753/29332732-e8243282-81b5-11e7-963d-30129a4a3737.png)

Looks like if you fight the battle, then the 'info' line pops up. Not sure why I saw it as a pop-up, opening the error console and then opening this save game and proceeding to battle should repro: 
[info_error.zip](https://github.com/triplea-game/triplea/files/1226158/info_error.zip)

**Comment by ssoloff** [comment]
Is the bug that the step name is missing or that the console popped up?

If the latter, #1950 already covers that issue.  Basically, anything going through a `Logger`, regardless of severity, will trigger the console because the default-configured `ConsoleHandler` sends all output to `stderr`.

**Comment by DanVanAtta** [comment]
Former, the info message is not expected.

**Comment by ron-murhammer** [comment]
@DanVanAtta @ssoloff This is closely related to the issues that I fixed in #1954. It has to do with the steps are all displayed in the battle window at the start of the round. Depending on casualties some of the withdraw options can change (think amphib units exist at beginning of the round but are all killed off during round so retreat option changes). I simplified the code to show a generic retreat option to avoid the changing retreat types. I guess there are still potential issues here though.

**Comment by simon33-2** [comment]
Excellent! This was one of the ones which I found annoying.

**Comment by simon33-2** [comment]
This is fixed in #1981.

Please close.

**Comment by ssoloff** [comment]
Closing per @simon33-2's confirmation.  @DanVanAtta, please reopen if there was something else you wanted fixed with this issue.
